Code/src/main/java/nl/utwente/viskell/ui/WireMenu.java

```diff
@@ -0,0 +1,102 @@
+package nl.utwente.viskell.ui;
+
+import javafx.animation.ScaleTransition;
+import javafx.scene.control.Button;
+import javafx.scene.layout.TilePane;
+import javafx.util.Duration;
+import nl.utwente.viskell.ui.components.*;
+
+/** A menu for attaching something to an open wire. */
+public class WireMenu extends TilePane {
+
+    /** The toplevel pane this menu is on. */
+    private final ToplevelPane toplevel;
+    
+    /** The draw wire belonging to this menu */
+    private DrawWire attachedWire;
+    
+    public WireMenu(DrawWire wire) {
+        this.toplevel = wire.getAnchor().getPane();
+        this.attachedWire = wire;
+        this.setMouseTransparent(true);
+
+        this.getStyleClass().add("menu");
+        this.setPrefColumns(1);
+        this.setVgap(5);
+        
+        Button cancelButton = new Button("Cancel");
+        cancelButton.setOnAction(event -> this.attachedWire.remove());
+        // FIXME: silly workaround, somehow the buttons in this menu don't work on touch click while those in FunctionMenu do...
+        cancelButton.setOnTouchPressed(event -> this.attachedWire.remove());
+        
+        if (wire.getAnchor() instanceof InputAnchor) {
+            Button lambdaBlockButton = new Button("Lambda");
+            lambdaBlockButton.setOnAction(event -> addBlockWithOutput(new DefinitionBlock(this.toplevel, 1)));
+            lambdaBlockButton.setOnTouchPressed(event -> addBlockWithOutput(new DefinitionBlock(this.toplevel, 1)));
+            
+            Button rationalBlockButton = new Button("Rational");
+            rationalBlockButton.setOnAction(event -> addBlockWithOutput(new SliderBlock(this.toplevel, false)));
+            rationalBlockButton.setOnTouchPressed(event -> addBlockWithOutput(new SliderBlock(this.toplevel, false)));
+            
+            Button IntegerBlockButton = new Button("Integer");
+            IntegerBlockButton.setOnAction(event -> addBlockWithOutput(new SliderBlock(this.toplevel, true)));
+            IntegerBlockButton.setOnTouchPressed(event -> addBlockWithOutput(new SliderBlock(this.toplevel, true)));
+
+            this.getChildren().addAll(cancelButton, lambdaBlockButton, rationalBlockButton, IntegerBlockButton);
+            
```

Comment:
More things should be added to this menu later, but I skipped ValueBlock because of prompting dialog breaks the nice local flow of using the wire menu.